src/OpenLoco/src/GameCommands/VehicleChangeRunningMode.cpp

```diff
@@ -0,0 +1,156 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Ui/WindowManager.h"
+#include "Vehicles/Vehicle.h"
+#include "World/CompanyManager.h"
+#include "World/StationManager.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004B6B0C
+    static bool sub_4B6B0C(Vehicles::VehicleHead* head)
+    {
+        // Looks like a variant of Vehicles::VehicleHead::canBeModified?
+        registers regs;
+        regs.esi = X86Pointer(head);
+        call(0x004B6B0C, regs);
+        return regs.eax != 0;
+    }
+
+    static void invalidateWindow(EntityId headId)
+    {
+        Ui::WindowManager::invalidate(Ui::WindowType::vehicle, static_cast<Ui::WindowNumber_t>(headId));
+    }
+
+    // 0x004B6AEE
+    static uint32_t changeLocalExpressMode(const Vehicles::Vehicle& train, const uint8_t flags)
+    {
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        train.veh1->var_48 ^= Vehicles::Flags48::expressMode;
+        invalidateWindow(train.head->head);
+        return 0;
+    }
+
+    // 0x004B6A08
+    static uint32_t startStopVehicle(const Vehicles::Vehicle& train, bool startVehicle, const uint8_t flags)
+    {
+        // Starting this vehicle -- can we?
+        if (startVehicle && !sub_4B6B0C(train.head))
+        {
+            return FAILURE;
+        }
+
+        // Stopping this vehicle, but vehicle is already stopped?
+        if (!startVehicle && train.head->hasVehicleFlags(VehicleFlags::commandStop))
+        {
+            return 0;
+        }
+
+        // Starting this vehicle, but vehicle is already travelling?
+        if (startVehicle && !train.head->hasVehicleFlags(VehicleFlags::commandStop))
+        {
+            return 0;
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        train.head->vehicleFlags ^= VehicleFlags::commandStop;
+        if (train.head->hasVehicleFlags(VehicleFlags::commandStop))
+        {
+            train.head->vehicleFlags &= ~VehicleFlags::manualControl;
+        }
+
+        if (!(train.head->hasVehicleFlags(VehicleFlags::unk_2) || !CompanyManager::isPlayerCompany(train.head->owner)))
+        {
+            auto madeProfit = train.veh2->profit[0] | train.veh2->profit[1] | train.veh2->profit[2] | train.veh2->profit[3];
+            if (madeProfit != 0)
+            {
+                StationManager::sub_437F29(getUpdatingCompanyId(), 1);
+            }
+        }
+
+        invalidateWindow(train.head->head);
+        return 0;
+    }
+
+    // 0x004B6AAF
+    static uint32_t toggleManualDriving(const Vehicles::Vehicle& train, const uint8_t flags)
+    {
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        train.head->vehicleFlags ^= VehicleFlags::manualControl;
+        train.head->vehicleFlags |= VehicleFlags::commandStop;
```

Comment:
This is only set if leaving manual control and cleared if entering manual control.

Reply:
Right, this was intended to be the equivalent of:
```
.text:004B6AB9                 bts     word ptr [esi+0Ch], 1
.text:004B6ABF                 bt      word ptr [esi+0Ch], 6
.text:004B6AC5                 jnb     short loc_4B6ACD
.text:004B6AC7                 btr     word ptr [esi+0Ch], 1
```
What am I missing?

Reply:
Ah, I had interpreted it the other way around... I'll address it.